timeless/access_control/director_privileges.py

```diff
@@ -12,29 +11,35 @@ def has_privilege(method=None, resource=None, *args, **kwargs) -> bool:
 
 
 def __employee_access(method=None, *args, **kwargs):
-    """
-    @todo #175:30min Add role to the employee model and check that Director
-     can only access/modify employees that have role of manager, master or
-     interns.
-    """
     permitted, user = False, flask.g.get("user")
     employee_id = kwargs.get("employee_id")
     if user:
         if employee_id:
-            permitted = check_employee(employee_id, method, user)
+            permitted = check_rights(employee_id, user)
         else:
             permitted = True
     return permitted
 
 
-def check_employee(employee_id, method, user):
-    if employee_id == user.id and method == Method.READ:
+def check_rights(employee_id, user):
+    """
+    Check the rights a director has over an employee profile.
+    In principle, a director can access their own profile and the
+    profile of the employees with a lower role, who work at the same company.
+    """
+    if employee_id == user.id:
         return True
     else:
         other = Employee.query.get(employee_id)
-        return user.company_id == other.company_id
+        other_role = other.role.name
+        return user.company_id == other.company_id and\
+            user.role.name == "director" and\
```

Comment:
@amihaiemil I think should hardcode all names directly in `Role` model (see https://github.com/timelesslounge/timelessis/blob/master/timeless/roles/models.py#L22). I left a puzzle there to find a nice way to handle role types, and I think while it's not resolved we should stay all hardcoded names there in the model. in order to make it easier to change in the future without searching any other places where it's used. wdyt?